libs/core/langchain_core/callbacks/manager.py

```diff
@@ -1494,6 +1493,39 @@ def on_retriever_start(
             inheritable_metadata=self.inheritable_metadata,
         )
 
+    def on_adhoc_event(
```

Comment:
Will be fun to explain the difference of between an adhoc event and an event within a run

Reply:
Let me know if you want to discuss -- we can have the tracer convert the adhoc event to an event?

Reply:
`adhoc` -> `custom` or `generic`?